packages/components/src/mobile/global-styles-context/utils.native.js

```diff
@@ -348,8 +348,8 @@ export function getColorsAndGradients(
 							},
 					  }
 					: features?.color ),
-				defaultPalette: !! defaultEditorColors,
-				defaultGradients: !! defaultEditorGradients,
+				defaultPalette: defaultEditorColors?.length > 0,
+				defaultGradients: defaultEditorGradients?.length > 0,
```

Comment:
Nitpick: Not sure if it's required that we guarantee that these values are always boolean type. If so, we could do:
```suggestion
				defaultPalette: defaultEditorColors?.length > 0 ?? false,
				defaultGradients: defaultEditorGradients?.length > 0 ?? false,
```

Reply:
Another option would be to set a default value to both arguments in order to assure that they're always defined:
```js
export function getColorsAndGradients(
	defaultEditorColors = [],
	defaultEditorGradients = [],
	rawFeatures
) {
```

Reply:
Thanks for the suggestion! I think I like the idea of adding a default value better. I'll do that and merge 🚀

Reply:
Added in https://github.com/WordPress/gutenberg/pull/38417/commits/01a8ad9d1fb22664f7c6d4fc7f86dbb9e231e3b4